pkg/filters/proxy/loadbalance.go

```diff
@@ -38,53 +41,126 @@ const (
 	LoadBalancePolicyIPHash = "ipHash"
 	// LoadBalancePolicyHeaderHash is the load balance policy of HTTP header hash.
 	LoadBalancePolicyHeaderHash = "headerHash"
+	// LoadBalanceStickyCookie is the generated cookie name used for sticky session
+	LoadBalanceStickyCookie = "EASEGRESS_SESSION"
 )
 
 // LoadBalancer is the interface of an HTTP load balancer.
 type LoadBalancer interface {
 	ChooseServer(req *httpprot.Request) *Server
+	Manipulate(server *Server, req *httpprot.Request, resp *httpprot.Response)
+	GetSlots() []*Server
 }
 
 // LoadBalanceSpec is the spec to create a load balancer.
 type LoadBalanceSpec struct {
 	Policy        string `json:"policy" jsonschema:"omitempty,enum=,enum=roundRobin,enum=random,enum=weightedRandom,enum=ipHash,enum=headerHash"`
 	HeaderHashKey string `json:"headerHashKey" jsonschema:"omitempty"`
+	Sticky        bool   `json:"sticky" jsonschema:"omitempty"`
+	StickyCookie  string `json:"stickyCookie" jsonschema:"omitempty"`
+	StickyExpire  string `json:"stickyExpire" jsonschema:"omitempty,format=duration"`
 }
 
 // NewLoadBalancer creates a load balancer for servers according to spec.
-func NewLoadBalancer(spec *LoadBalanceSpec, servers []*Server) LoadBalancer {
+func NewLoadBalancer(spec *LoadBalanceSpec, servers []*Server, slots []*Server) LoadBalancer {
+	d, err := time.ParseDuration(spec.StickyExpire)
+	if err != nil {
+		logger.Errorf("failed to parse duration: %s", spec.StickyExpire)
+	}
+	if spec.StickyCookie == LoadBalanceStickyCookie {
+		logger.Errorf("can't use system cookie name: %s", LoadBalanceStickyCookie)
+	}
```

Comment:
question: why do we keep processing when we encounter the error? if we want to process with the errors, it's better to use the default value.